lib/validation.js

```diff
@@ -22,12 +22,27 @@ function compileSchemasForSerialization (context, compile) {
         throw new Error('response schemas should be nested under a valid status code, e.g { 2xx: { type: "object" } }')
       }
 
-      acc[statusCode] = compile({
-        schema,
-        url,
-        method,
-        httpStatus: statusCode
-      })
+      if (Array.isArray(schema)) {
+        const contentTypesSchemas = {}
+        schema.forEach(property => {
```

Comment:
we can use a `.reduce` here

Reply:
a for loop is even better

Reply:
I will go with for loop to avoid having nested reduce calls